thorlcr/msort/tsortl.cpp

```diff
@@ -298,6 +306,12 @@ ISocketRowWriter *ConnectMergeWrite(IThorRowInterfaces *rowif,ISocket *socket,si
     TransferStreamHeader hdr;
     socket->read(&hdr,sizeof(hdr));
```

Comment:
What happens if less than sizeof(hdr) bytes are available?

Reply:
not new, but it would block indefinitely until/if ever, it got sizeof(hdr)
It could be changed to readtms, and specify size(hdr) as max, and timeout if gets less over a short period.

Reply:
You would want min == max == sizeof(hdr) with a timeout.
Even if sizeof(hdr) is small you cannot always guarantee all the msg arrives at once.

Reply:
You could use min == size of the crc if it were the first item of hdr and then continue on with an addl read if ok and don't have it all.

Reply:
To guard against a 'rogue' client connecting and sending less than sizeof(hdr),
you want min to be 1, and loop until read sizeof(hdr), but timeout once started reading.
If it's a genuine client, it should send all swiftly once there is anything on the wire.

If it's a foreign client sending a single byte, then it should timeout after a short time, and reject it.

Reply:
Yes, a loop until done is the ideal way.